lib/core/src/lib/datatable/components/datatable/datatable.component.ts

```diff
@@ -527,9 +528,15 @@ export class DataTableComponent implements OnInit, AfterContentInit, OnChanges,
     private handleRowSelection(row: DataRow, e: KeyboardEvent | MouseEvent) {
         if (this.data) {
             if (this.isSingleSelectionMode()) {
-                this.resetSelection();
-                this.selectRow(row, true);
-                this.emitRowSelectionEvent('row-select', row);
+                const isRowSelected = row.isSelected
+                if (isRowSelected) {
```

Comment:
If you use `isRowSelected` only in the if, you can just use `row.isSelected` directly without creating new variable